flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpoint.java

```diff
@@ -290,16 +289,14 @@ public static boolean checkpointsMatch(
 		Collection<CompletedCheckpoint> first,
 		Collection<CompletedCheckpoint> second) {
 
-		Set<Tuple2<Long, JobID>> firstInterestingFields =
-			new HashSet<>();
+		List<Tuple2<Long, JobID>> firstInterestingFields = new ArrayList<>();
 
 		for (CompletedCheckpoint checkpoint : first) {
```

Comment:
For stability purposes we may want to check whether the input collections are actually sorted.. I know that the sole user `ZooKeeperCompletedCheckpointStore` does in fact pass sorted lists, but technically there's no guarantee at the moment.

Reply:
They shouldn't be sorted, they should just have the order that they have in ZooKeeper.